staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -113,6 +116,11 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			if v.Name == version.Version {
 				foundThisVersion = true
 			}
+			if v.Storage {
+				bytes := sha256.Sum256([]byte(gv.Group + "/" + gv.Version + "/" + crd.Spec.Names.Kind))
+				storageVersionHash = base64.StdEncoding.EncodeToString(
+					bytes[:8])
```

Comment:
Please add the math you did as a comment for future readers.

Let's plan for 10k (10000) types and < 1 / billion false positive rate, which is ridiculously over powered :)

Reply:
Done.